app/error-navigation.js

```diff
@@ -0,0 +1,48 @@
+var dashify = require('./templates/dashify.js');
+var $ = require('jquery');
+
+var errorNavigation;
+
+function resetErrorNavigation(){
+  errorNavigation = {
+    index: 0,
+    errors: []
+  }
+};
+resetErrorNavigation();
+
+function registerErrors(elist){
+  errorNavigation.errors = elist;
+}
+
+function scrollToError(){
+    $("#"+dashify(errorNavigation.errors[errorNavigation.index].location))[0]
+    .scrollIntoView();
+}
+
+function moveError(by){
+  var i = errorNavigation.index;
+  var len = errorNavigation.errors.length;
+  var next = (((i + by) % len) + len) % len; // ensure positive, bounded
+
+  errorNavigation.index = next;
+  scrollToError();
+}
+
+document.addEventListener("keydown", function(e){
```

Comment:
As much as possible, I like to enforce side-effect-free imports, so this event listener would be better in main.js or in some kind of `init()` method that gets exported.